launcher/container_runner.go

```diff
@@ -194,43 +208,112 @@ func (r *ContainerRunner) measureContainerClaims(ctx context.Context) error {
 		}
 	}
 
-	contianerSpec, err := r.container.Spec(ctx)
+	containerSpec, err := r.container.Spec(ctx)
 	if err != nil {
 		return err
 	}
-	for _, arg := range contianerSpec.Process.Args {
+	for _, arg := range containerSpec.Process.Args {
 		if err := r.attestAgent.MeasureEvent(cel.CosTlv{EventType: cel.ArgType, EventContent: []byte(arg)}); err != nil {
 			return err
 		}
 	}
-	for _, env := range contianerSpec.Process.Env {
+	for _, env := range containerSpec.Process.Env {
 		if err := r.attestAgent.MeasureEvent(cel.CosTlv{EventType: cel.EnvVarType, EventContent: []byte(env)}); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// Run the container
-// Doesn't support container restart yet
-// Container output will always be redirected to stdio for now
-func (r *ContainerRunner) Run(ctx context.Context) error {
-	if err := r.measureContainerClaims(ctx); err != nil {
-		return fmt.Errorf("failed to measure container claims: %v", err)
+func getTTL(token []byte) (time.Duration, error) {
+	claims := &jwt.RegisteredClaims{}
+	_, _, err := new(jwt.Parser).ParseUnverified(string(token), claims)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse token: %w", err)
+	}
+
+	now := time.Now()
+	if !now.Before(claims.ExpiresAt.Time) {
+		return 0, errors.New("token is expired")
+	}
+
+	return time.Until(claims.ExpiresAt.Time), nil
+}
+
+func (r *ContainerRunner) refreshToken(ctx context.Context, refreshMultiplier float64) (time.Duration, error) {
+	if refreshMultiplier > 1 {
```

Comment:
let's store this in the tokenRefresher struct and check it once when setting it.